app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetRecordController.scala

```diff
@@ -22,142 +22,163 @@
         }
       }
-    }
   }
 
-  def save(id: String, oci: String): Action[AnyContent] = Action.async { implicit request: Request[AnyContent] =>
+  private def submit(
+    editSetId: String,
+    editSetRecord: EditSetRecord,
+    editSetRecordFormValues: EditSetRecordFormValues
+  ): Future[Result] =
+    editRecordSetService
+      .updateEditSetRecord(editSetId, editSetRecord.oci, editSetRecordFormValues)
+      .unsafeToFuture()
+      .map { updateResponseStatus =>
+        logger
+          .info(s"We are currently ignoring the returned status of the update: [$updateResponseStatus]")
+        Redirect(controllers.routes.EditSetRecordController.save(editSetId, editSetRecord.oci))
+      }
+
+  def save(editSetId: String, oci: String): Action[AnyContent] = Action.async { implicit request: Request[AnyContent] =>
     withUserAsync { user =>
-      findRecord(id, oci).map {
-        case Right(record) =>
-          val title = resolvedMessage(MessageKeys.title)
-          val editSetName = editSets.getEditSet().name
-          val heading = resolvedMessage(MessageKeys.heading, record.ccr)
-          val message = resolvedMessage(MessageKeys.buttonSave)
-          val recordType = record.recordType
-          logger.info(s"Save changes for record id $oci edit set id $id")
-          Ok(editSetRecordEditSave(user, editSetName, title, heading, oci, message, recordType))
-
-        case Left(_) => NotFound
+      findEditSet(editSetId).flatMap {
+        case Right(editSet) =>
+          findEditSetRecord(editSetId, oci).map {
+            case Right(editSetRecord) => generateSaveView(user, editSet, editSetRecord)
+            case Left(_) => NotFound(s"Edit Set Record [$editSetId] for Edit Set [$editSetId] was not found")
+          }
+        case Left(_) => NotFound(s"Edit Set [$editSetId] was not found")
       }
     }
   }
 
-  def discard(id: String, oci: String): Action[AnyContent] = Action.async { implicit request: Request[AnyContent] =>
-    withUserAsync { user =>
-      findRecord(id, oci).map {
-        case Right(record) =>
-          val title = resolvedMessage(MessageKeys.title)
-          val editSetName = editSets.getEditSet().name
-          val heading = resolvedMessage(MessageKeys.heading, record.ccr)
-          val message = resolvedMessage(MessageKeys.buttonDiscard)
-          val recordType = record.recordType
-          logger.info(s"Discard changes for record id $oci edit set id $id ")
-
-          Ok(editSetRecordEditDiscard(user, editSetName, title, heading, oci, message, recordType))
-        case Left(_) => NotFound
+  def discard(editSetId: String, oci: String): Action[AnyContent] = Action.async {
+    implicit request: Request[AnyContent] =>
+      withUserAsync { user =>
+        findEditSet(editSetId).flatMap {
+          case Right(editSet) =>
+            findEditSetRecord(editSetId, oci).map {
+              case Right(editSetRecord) => generateDiscardView(user, editSet, editSetRecord)
```

Comment:
I think we could use a for comprehension here rather than flatMap followed by map.

What do you think about creating a utility method in the base controller to get both the edit set and edit set record? It's a pattern we're repeating a lot here with very similar semantics.

Reply:
I'll try that.

As for the utility method, I could see that being useful in this class, as it's only here that we fetch the pair of them.

Reply:
Resolving as @felixgb approved the PR.